services/dynamic/dynamic-xml.tester.js

```diff
@@ -25,76 +28,95 @@ t.create('No query specified')
 
 t.create('XML from url')
   .get(
-    '.json?url=https://services.addons.mozilla.org/en-US/firefox/api/1.5/addon/707078&query=/addon/name'
+    `.json?${queryString.stringify({
+      url: exampleUrl,
+      query: "//book[@id='bk102']/title",
+    })}`
   )
+  .intercept(withExampleXml)
   .expectBadge({
     label: 'custom badge',
-    message: 'IndieGala Helper',
+    message: 'Midnight Rain',
     color: 'blue',
   })
 
-t.create('XML from uri (support uri query parameter)')
+t.create('uri query parameter alias')
   .get(
-    '.json?uri=https://services.addons.mozilla.org/en-US/firefox/api/1.5/addon/707078&query=/addon/name'
+    `.json?${queryString.stringify({
+      uri: exampleUrl,
+      query: "//book[@id='bk102']/title",
+    })}`
   )
+  .intercept(withExampleXml)
   .expectBadge({
     label: 'custom badge',
-    message: 'IndieGala Helper',
+    message: 'Midnight Rain',
     color: 'blue',
   })
 
-t.create('XML from url (attribute)')
+t.create('attribute')
   .get(
-    '.json?url=https://services.addons.mozilla.org/en-US/firefox/api/1.5/addon/707078&query=/addon/reviews/@num'
+    `.json?${queryString.stringify({
+      url: exampleUrl,
+      query: '//book[2]/@id',
+    })}`
   )
+  .intercept(withExampleXml)
   .expectBadge({
     label: 'custom badge',
-    message: Joi.string().regex(/^\d+$/),
+    message: 'bk102',
   })
 
-t.create('XML from url | multiple results')
+t.create('multiple results')
   .get(
-    '.json?url=https://services.addons.mozilla.org/en-US/firefox/api/1.5/addon/707078&query=/addon/compatible_applications/application/name'
+    `.json?${queryString.stringify({
+      url: exampleUrl,
+      query: '//book/title',
+    })}`
   )
+  .intercept(withExampleXml)
   .expectBadge({
     label: 'custom badge',
-    message: Joi.string().regex(
-      /^Firefox( for Android)?,\sFirefox( for Android)?$/
-    ),
+    message:
+      "XML Developer's Guide, Midnight Rain, Maeve Ascendant, Oberon's Legacy, The Sundered Grail, Lover Birds, Splish Splash, Creepy Crawlies, Paradox Lost, Microsoft .NET: The Programming Bible, MSXML3: A Comprehensive Guide, Visual Studio 7: A Comprehensive Guide",
   })
 
-t.create('XML from url | caching with new query params')
```

Comment:
I removed this test ^^ Frankly I have no idea what it is for. It looks like it was copied and pasted, or modified into oblivion.